pkg/provider/provider.go

```diff
@@ -42,6 +43,9 @@ import (
 
 // KedaProvider implements External Metrics Provider
 type KedaProvider struct {
+	defaults.DefaultCustomMetricsProvider
```

Comment:
that is quite surprising to me. I wouldn't expect you to have to implement this provider since you are not using it at all in keda

Reply:
We have to implement the whole Provider, maybe we are doing it wrongly, and we could  implement just the external metrics part. @zroubalik do you know something about this?

Reply:
Damien is right, external metrics are enough. We should chage the return value of this:
https://github.com/kedacore/keda/blob/9d22b0c510f971cfaa99d43e4ed2388b78d83423/pkg/provider/provider.go#L64

to `ExternalMetricsProvider`

Reply:
You are totally right, that's enough